aten/src/ATen/native/cuda/PersistentSoftmax.cuh

```diff
@@ -183,16 +188,6 @@ __global__ void softmax_warp_forward(output_t *dst, const input_t *src, int batc
         for (int it = 0;  it < WARP_ITERATIONS;  ++it) {
             int element_index = local_idx + it * WARP_SIZE;
             if (element_index < element_count) {
-                if (is_masked) {
```

Comment:
So this code was just entirely unused or is it a change that still makes the tests pass?

Reply:
this path forcibly wrote zeros to output values. instead of doing that, we update the temporary `elements` array, and then let line 194 (previously 199) below produce a NaN if the entire row was masked.

Reply:
This is fine if NaNs don't cause performance regressions. We can expect to see users mask out entire rows. @ngimel can I hand this off to you?

Reply:
NaNs are the required behavior for compatibility with the existing softmax implementation.